Formula/gst-python.rb

```diff
@@ -10,21 +10,50 @@ class GstPython < Formula
     sha256 "e708aa675d5744fc36608249728290d6d97803cd071bf2ac91d245a9d8e8c996" => :mavericks
   end
 
+  option "without-python"
+
+  depends_on :python3 => :optional
   depends_on "gst-plugins-base"
-  depends_on "pygobject3"
+
+  requires_py3 = []
+  requires_py3 << "with-python3" if build.with? "python3"
+  depends_on "pygobject3" => requires_py3
 
   link_overwrite "lib/python2.7/site-packages/gi/overrides"
 
   def install
-    # pygi-overrides-dir switch ensures files don't break out of sandbox.
-    system "./configure", "--disable-dependency-tracking",
-                          "--disable-silent-rules",
-                          "--prefix=#{prefix}",
-                          "--with-pygi-overrides-dir=#{lib}/python2.7/site-packages/gi/overrides"
-    system "make", "install"
+    if build.with?("python") && build.with?("python3")
+      # Upstream does not support having both Python2 and Python3 versions
+      # of the plugin installed because apparently you can load only one
+      # per process, so GStreamer does not know which to load.
+      odie "Options --with-python and --with-python3 are mutually exclusive."
+    end
+
+    Language::Python.each_python(build) do |python, version|
+      # pygi-overrides-dir switch ensures files don't break out of sandbox.
+      args = %W[
+        --disable-dependency-tracking
+        --disable-silent-rules
+        --prefix=#{prefix}
+        --with-pygi-overrides-dir=#{lib}/python#{version}/site-packages/gi/overrides
+        PYTHON=#{python}
+      ]
+
+      if python == "python3"
+        python_framework = Formula["python3"].opt_prefix/"Frameworks/Python.framework/Versions/#{version}"
+        args << "--with-libpython-dir=#{python_framework}/lib"
+      end
+
+      system "./configure", *args
+      system "make", "clean"
+      system "make", "install"
```

Comment:
What is the `make clean` good for? If I'm not misreading the formula, it will only build bindings for a single Python version. If we wanted to retain it, wouldn't it make more sense to put it after `make install`?